Mono.Addins/Mono.Addins/ExtensionNode.cs

```diff
@@ -171,7 +171,14 @@ internal void SetData (AddinEngine addinEngine, string plugid, ExtensionNodeType
 					try {
 						value (this, new ExtensionNodeEventArgs (ExtensionChange.Add, node));
 					} catch (Exception ex) {
-						addinEngine.ReportError (null, node.Addin != null ? node.Addin.Id : null, ex, false);
+						RuntimeAddin nodeAddin;
+						try {
+							nodeAddin = node.Addin;
+						} catch (Exception addinException) {
+							addinEngine.ReportError (null, null, addinException, false);
+							nodeAddin = null;
+						}
+						addinEngine.ReportError (null, nodeAddin != null ? nodeAddin.Id : null, ex, false);
```

Comment:
This should still be in the try statement

Reply:
No it should not. Wouldn't work if node.Addin throws.